lesspipe.sh

```diff
@@ -437,23 +437,25 @@ has_colorizer () {
 	[[ -n $3 ]] && lang=$3 || lang=$2
 	case $prog in
 		bat|batcat)
+		        batconfig=$(bat --config-file)
 			[[ -n $lang ]] && $prog --list-languages|sed 's/.*:/,/;s/$/,/'|grep -i ",$lang," > /dev/null && opt=(-l "$lang")
 			[[ -n $LESSCOLORIZER && $LESSCOLORIZER = *\ *--style=* ]] && style="${LESSCOLORIZER/* --style=/}"
 			[[ -z $style ]] && style=$BAT_STYLE
 			[[ -n $LESSCOLORIZER && $LESSCOLORIZER = *\ *--theme=* ]] && theme="${LESSCOLORIZER/* --theme=/}"
 			[[ -z $theme ]] && theme=$BAT_THEME
-			if [[ -r "$HOME/.config/bat/config" ]]; then
+			if [[ -r "$batconfig" ]]; then
 				if [[ -z $style ]]; then
-					grep -q -e '^--style' "$HOME/.config/bat/config" || style=plain
+					grep -q -e '^--style' "$batconfig" || style=plain
 				fi
 				if [[ -z $theme ]]; then
-					grep -q -e '^--theme' "$HOME/.config/bat/config" || theme=ansi
+					grep -q -e '^--theme' "$batconfig" || theme=ansi
 				fi
 			else
 				[[ -z $style ]] && style=plain
 				[[ -z $theme ]] && theme=ansi
 			fi
-			opt+=(--style="${style%% *}" --theme="${theme%%[|&;<>]*}")
+			[[ -n $style ]] && opt+=(--style="${style%% *}")
+			[[ -n $theme ]] && opt+=(--theme="${theme%%[|&;<>]*}")
```

Comment:
`$style` and `${style%% *}` can be different, e.g., `style=' plain`.

```suggestion
			style="${style%% *}" theme="${theme%%[|&;<>]*}")
			opt+=("${style:+--style=$style}" "${theme:+--theme=$theme}")
```

Reply:
I'm not sure this addresses the issue I'm trying to fix.  I get what you are saying, but your suggestion also removes the not empty check.  I'll do some testing and report back.

Reply:
Ah, sorry, I didn't notice the `${style:+ ...}`